src/KernelSigner.ts

```diff
@@ -790,6 +793,41 @@ export class KernelSigner {
     } as TransactionReturnType;
   }
 
+  public async addStake(args: AddStake, waitForReceipt: boolean = true,
+                        optionalArgs?: OptionalArgs) {
+    const client = await this.getActiveClient();
+    const addStakeCallData = await addStake(args, client, this.config.environment);
+    const userOpHash = await this._sendUserOperation(client, addStakeCallData, optionalArgs);
+    if (waitForReceipt) {
+      const client = await this.getActiveClient();
```

Comment:
although its not all together impossible, it's pretty unlikely that a session would be valid on line 798 and invalid by the time it gets to this line, since the logic to open a new session has a buffer to ensure that it executes before old session expires. because of this, I think we can remove this line where we get the active client again